internal/cmd/kafka/command_topic_consume.go

```diff
@@ -144,6 +147,9 @@ func (c *command) consume(cmd *cobra.Command, args []string) error {
 	}
 
 	rebalanceCallback := GetRebalanceCallback(offset, partitionFilter)
+	if commitGroupOffset && !cmd.Flags().Changed("from-beginning") && !cmd.Flags().Changed("offset") {
+		rebalanceCallback = nil
```

Comment:
Do you mind explaining how the rebalance callback works? I thought this was related to when a cluster adds or subtracts partitions?

Reply:
The rebalanceCallback is a function that gets called when the Kafka client needs to rebalance its consumer group, for example when a new consumer joins the group, or an existing consumer leaves the group. it allows me to define a specific offset to consume from when user passes in `--offset`